multiplatform-markdown-renderer/src/commonMain/kotlin/com/mikepenz/markdown/annotator/AnnotatedStringKtx.kt

```diff
@@ -133,18 +133,37 @@ fun AnnotatedString.Builder.appendMarkdownLink(
         return
     }
     val text = linkText.firstOrNull()?.getUnescapedTextInNode(content)
-    val destination = node.findChildOfType(MarkdownElementTypes.LINK_DESTINATION)?.getUnescapedTextInNode(content)
-    val linkLabel = node.findChildOfType(MarkdownElementTypes.LINK_LABEL)?.getUnescapedTextInNode(content)
+    val destination =
+        node.findChildOfType(MarkdownElementTypes.LINK_DESTINATION)?.getUnescapedTextInNode(content)
+    val linkLabel =
+        node.findChildOfType(MarkdownElementTypes.LINK_LABEL)?.getUnescapedTextInNode(content)
     val annotation = destination ?: linkLabel
 
     if (annotation != null) {
-        if (text != null) annotatorSettings.referenceLinkHandler?.store(text, annotation)
-        withLink(LinkAnnotation.Url(annotation, annotatorSettings.linkTextSpanStyle, annotatorSettings.linkInteractionListener)) {
+        text?.let { annotatorSettings.referenceLinkHandler?.store(it, annotation) }
+
+        val linkStyle = if (node.parent?.type == GFMElementTypes.STRIKETHROUGH) {
```

Comment:
<img width="885" alt="Screenshot 2025-02-07 at 11 42 57" src="https://github.com/user-attachments/assets/3110f2c6-60ad-4286-908b-60d1652c8296" />

Looking more into this. parent wouldn't work unfortunately. as the link style may be surrounding nodes. 

I wonder if the problem should be mostly considered to be they way `TextLinkStyles` work in annotated strings :/